test/BlobStorageTest.t.sol

```diff
@@ -0,0 +1,164 @@
+        bytes memory buf = new bytes(length);
+        for (uint256 i = 0; i < length; i++) {
+            buf[i] = bytes1(val);
+        }
+        return buf;
+    }
+
+    function testPutAndGet() public {
+        uint256 fileSize = BLOB_SIZE * 2 + 1024;
+        bytes memory fileData = createBytes(fileSize, 0x01);
+
+        uploadFile(fileData);
+
+        (uint256 actualSize, ) = ercBlob.size(hexName);
+        assertEq(actualSize, fileSize);
+
+        uint256 chunkCount = ercBlob.countChunks(hexName);
+        assertEq(chunkCount, 3);
+
+        // chunk size
+        (uint256 chunkSize0, ) = ercBlob.chunkSize(hexName, 0);
+        assertEq(chunkSize0, BLOB_SIZE);
+
+        (uint256 chunkSize2, ) = ercBlob.chunkSize(hexName, 2);
+        assertEq(chunkSize2, fileSize - BLOB_SIZE * 2);
+
+        // chunk hash
+        bytes32 chunkHash0 = ercBlob.getChunkHash(hexName, 0);
+        bytes32 localHash0 = createLocalHash(BLOB_SIZE);
+        assertEq(chunkHash0, localHash0);
+
+        bytes32 chunkHash2 = ercBlob.getChunkHash(hexName, 2);
+        bytes32 localHash2 = createLocalHash(fileSize - BLOB_SIZE * 2);
+        assertEq(chunkHash2, localHash2);
+
+        // file hash
+        (bytes memory outData, ) = ercBlob.read(hexName);
+        assertEq(keccak256(outData), keccak256(fileData));
+    }
+
+    function testRemove() public {
+        uint256 fileSize = BLOB_SIZE * 4;
+        bytes memory fileData = createBytes(fileSize, 0x01);
+
+        uploadFile(fileData);
+
+        // Remove last chunk
+        ercBlob.removeChunk(hexName, 3);
+        (uint256 sizeAfterRemove, ) = ercBlob.size(hexName);
+        assertEq(sizeAfterRemove, BLOB_SIZE * 3);
+
+        // Truncate to 1
+        ercBlob.truncate(hexName, 1);
+        (uint256 sizeAfterTruncate, ) = ercBlob.size(hexName);
+        assertEq(sizeAfterTruncate, BLOB_SIZE);
+
+        // Remove all
+        ercBlob.remove(hexName);
+        (uint256 finalSize, ) = ercBlob.size(hexName);
+        assertEq(finalSize, 0);
+    }
+
+    // === Utility ===
+    function uploadFile(bytes memory fileData) internal {
+        uint256 numChunks = (fileData.length + BLOB_SIZE - 1) / BLOB_SIZE;
+
+        uint256[] memory chunkIds = new uint256[](numChunks);
+        uint256[] memory chunkSizes = new uint256[](numChunks);
+        bytes[] memory chunkDatas = new bytes[](numChunks);
+        for (uint256 i = 0; i < numChunks; i++) {
+            chunkIds[i] = i;
+            uint256 start = i * BLOB_SIZE;
+            uint256 end = (start + BLOB_SIZE > fileData.length) ? fileData.length : (start + BLOB_SIZE);
+            chunkSizes[i] = end - start;
+            chunkDatas[i] = slice(fileData, start, end);
+        }
+
+        vm.deal(address(this), 1 ether);
+        uint256 cost = ercBlob.upfrontPayment() * numChunks;
+        vm.recordLogs();
+
+        ercBlob.writeChunksByBlobs{value: cost}(hexName, chunkIds, chunkSizes);
```

Comment:
Should any test cases be added for `writeChunkByCalldata`?

Reply:
LargeStorageTest.t.sol has been implemented